bottlerocket/agents/src/bin/vsphere-k8s-cluster-resource-agent/vsphere_k8s_cluster_provider.rs

```diff
@@ -251,6 +278,7 @@ async fn create_vsphere_k8s_cluster(
     bundle_manifest_path: &str,
     resources: &mut Resources,
     memo: &mut ProductionMemo,
+    secret_manager_client: &aws_sdk_secretsmanager::Client,
```

Comment:
You've tied our function interface to the AWS SDK's client. This might be fine but can lead to problems if they change their interface it breaks out contract with the function API or we want to switch to a different backing for this token. It would probably be cleaner to provide a `ClusterTokensClient` that happens to wrap secrets manager right now, but would allow us to switch to a different implementation without changing the function signature.